mongoengine/fields.py

```diff
@@ -1622,6 +1623,57 @@ def prepare_query_value(self, op, value):
         return super().prepare_query_value(op, self.to_mongo(value))
 
 
+class EnumField(BaseField):
+    """ Enumeration Field. Values are stored underneath as strings.
+     Example usage:
+    .. code-block:: python
+
+        class Status(Enum):
+            NEW = 'new'
+            DONE = 'done'
+
+        class ModelWithEnum(Document):
+            status = EnumField(Status, default=Status.NEW)
+
+        ModelWithEnum(status='done')
+        ModelWithEnum(status=Status.DONE)
+
+    Enum fields can be searched using enum or its value:
+    .. code-block:: python
+
+        ModelWithEnum.objects(status='new').count()
+        ModelWithEnum.objects(status=Status.NEW).count()
+    """
+    def __init__(self, enum, **kwargs):
+        self._enum_cls = enum
+        kwargs["choices"] = list(self._enum_cls)
```

Comment:
This will silently drop users provided "choices" which may lead to confusion. I think it's safer to check if "choices" is in kwargs and if yes raise an exception like `ValueError("'choices' can't be set on EnumField, it is implicitly set as the enum class")`